kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ResourcesUtil.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.stream.Collector;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.OwnerReference;
+import io.fabric8.kubernetes.api.model.OwnerReferenceBuilder;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+import io.kroxylicious.kubernetes.api.v1alpha1.kafkaproxyspec.Clusters;
+
+public class ResourcesUtil {
+    private ResourcesUtil() {
+    }
+
+    static <O extends HasMetadata> OwnerReference ownerReferenceTo(O owner) {
+        return new OwnerReferenceBuilder()
+                .withKind(owner.getKind())
+                .withApiVersion(owner.getApiVersion())
+                .withName(owner.getMetadata().getName())
+                .withUid(owner.getMetadata().getUid())
+                .build();
+    }
+
+    static List<Clusters> distinctClusters(KafkaProxy primary) {
```

Comment:
I'd much rather this was a method on `KafkaProxy` or we had a wrapper type around the KafkaProxy to provide this rather some random collection of Util methods...

Reply:
`KafkaProxy` is generated, so we can't go adding helper functions to it. I suspect this method will disappear soon once we work more on how to report per-cluster errors. So I think this is OK for now.